test/assert.js

```diff
@@ -182,6 +183,29 @@ test('.throws()', function (t) {
 	t.end();
 });
 
+test('.throws() becomes the thrown Error', function (t) {
+	var expected = new Error();
+	var actual = assert.throws(function () {
+		throw expected;
+	});
+
+	assert.same(actual, expected);
+
+	t.end();
+});
+
+test('.throws() becomes the rejection reason of promise', function (t) {
+	var expected = new Error();
+	var exec = new Promise(function (resolve) {
+		resolve(assert.throws(Promise.reject(expected)));
+	});
+
+	exec.then(function (actual) {
```

Comment:
Gosh. I really do like to [overcomplicate all the things](https://github.com/kasperlewau/ava/commit/3f24a108f516cce747191a9a788c3c337e4a66e5) :disappointed: